components/style/stylist.rs

```diff
@@ -119,15 +128,26 @@ impl DocumentCascadeData {
                     Origin::UserAgent | Origin::User
                 ));
 
-                if origins_to_rebuild.contains(sheet_origin.into()) {
-                    self.add_stylesheet(
-                        device,
-                        quirks_mode,
-                        stylesheet,
-                        guards.ua_or_user,
-                        extra_data,
-                    );
+                let validity = flusher.origin_validity(sheet_origin);
+
+                // Servo doesn't support to incrementally mutate UA sheets.
```

Comment:
To reinforce this, would it make sense to assert in here that if ua_stylesheets is non-None that this is not stylo?